### PR TITLE
Fix issue #1573

### PR DESCRIPTION
**Pull request by vayunbiyani**
Added window height event listener to update the height. Implemented a debounce to avoid unnecessary overheads

### CODE REVIEW

File: quadratic-client/src/app/ui/menus/CodeEditor/panels/CodeEditorPanelSide.tsx [hunk 1]
```diff
@@ -6,6 +6,17 @@ import { CodeEditorPanelData } from '@/app/ui/menus/CodeEditor/panels/useCodeEdi
 import { useCallback, useEffect, useMemo, useState } from 'react';
 import { ResizeControl } from './ResizeControl';
 
+function debounce<T extends (...args: any[]) => void>(func: T, wait: number): T {
```
Comment (ddimaria) [inline quadratic-client/src/app/ui/menus/CodeEditor/panels/CodeEditorPanelSide.tsx:9]: There is a useDebounce hook located at `quadratic-client/src/shared/hooks/useDebounce.tsx`

Comment (vayunbiyani) [reply quadratic-client/src/app/ui/menus/CodeEditor/panels/CodeEditorPanelSide.tsx:9]: Awesome. I'll implement it shortly

**Comment by vayunbiyani** [comment]
> We require contributors to sign our Contributor License Agreement, and we don"t have one on file for you. In order for us to review and merge your code, please contact our team at https://www.quadratichq.com/contact.

@ddimaria This is resolved.

**Comment by vayunbiyani** [comment]
Hey @ddimaria @luke-quadratic @AyushAgrawal-A2 , just wanted to check in. Did you have a chance to review this

**Comment by davidkircos** [comment]
Hey, @vayunbiyani, I sent you an email to get the Contributor License Agreement signed

**Pull request closed without merging**